src/wallet/wallet.cpp

```diff
@@ -2467,8 +2467,11 @@ bool CWallet::OutputEligibleForSpending(const COutput& output, const CoinEligibi
     if (output.nDepth < (output.tx->IsFromMe(ISMINE_ALL) ? eligibility_filter.conf_mine : eligibility_filter.conf_theirs))
         return false;
 
-    if (!mempool.TransactionWithinChainLimit(output.tx->GetHash(), eligibility_filter.max_ancestors, eligibility_filter.max_descendants))
+    int64_t ancestors, descendants;
+    mempool.GetTransactionAncestry(output.tx->GetHash(), ancestors, descendants);
+    if (ancestors >= eligibility_filter.max_ancestors || descendants >= eligibility_filter.max_descendants) {
```

Comment:
commit: Switch to GetTransactionAncestry() in OutputEligibleForSpending

I think these should be strict `>` checks, in which case you don't need to switch to signed results. This makes sense logically because ancestors and descendants counts includes the transaction queried, so if the count is at the maximum allowable, it is still valid to create one more in the chain.

Reply:
It means having to change the parameter for all the calls (bump by 1), but I think you're right. The -1 hack is bad. Fixing.

Reply:
I had to do one minor tweak, but otherwise the current values worked even with the change. I also switched from `nMaxChainLength` to using `max_ancestors` and `max_descendants` with their corresponding defaults. (See 6ab847bce76224ff46552dd647b542fa70350a91.)